modules/core/src/mathfuncs_core.simd.hpp

```diff
@@ -224,6 +226,40 @@ void magnitude32f(const float* x, const float* y, float* mag, int len)
     }
 }
 
+void magnitude32fc(const float* xy, float* mag, int len)
+{
+    CV_INSTRUMENT_REGION();
+
+    int i = 0;
+
+#if CV_SIMD
+    const int VECSZ = v_float32::nlanes;
+    for( ; i < len; i += VECSZ*2 )
```

Comment:
Why is loop unrolled? Are there benefits for that?

Reply:
I just took the cv::magnitude implementation.
I assumed that it was designed as a hint to allow specific compiler optimization on 256b registers.